File: flutter_ads_native/android/src/main/kotlin/com/example/flutter_ads_native/tracking/AnalyticsEventNames.kt
package com.example.flutter_ads_native.tracking

/**
 * Centralized analytics event names for the local ads plugin.
 *
 * Keep names short, stable and snake_case to work well with Firebase/GA4.
 */
object AnalyticsEventNames {
    // Ads
    const val AD_LOAD = "swc_ad_load" // attempt/result of load
    const val AD_SHOW_CALL = "swc_ad_show_call" // show() invoked from Flutter
    const val AD_IMPRESSION = "swc_ad_impression" // ad actually displayed (impression)
    const val AD_SHOW_FAIL = "swc_ad_show_fail" // failed to show full screen ad
}

File: flutter_ads_native/ios/Classes/Tracking/AnalyticsEventNames.swift
import Foundation

// Centralized analytics event names for the local ads plugin (iOS side).
// Note: iOS ads implementation isn't wired in this repo yet; these constants
// exist to keep event naming consistent across platforms.
enum AnalyticsEventNames {
  static let adLoad = "swc_ad_load"
  static let adShowCall = "swc_ad_show_call"
  static let adImpression = "swc_ad_impression"
  static let adShowFail = "swc_ad_show_fail"
}


